### DISCUSSION TITLE
Custom CI webhook trigger

### DISCUSSION THREAD

**Discussion by kyeshmz** [codemagic-ci-cd/codemagic-docs#906] [Q&A] [answered]
Hi,

I was wondering if there was a way to set off the automatic webhook triggering from inside a CI script rather than handling it with the git managers, ie github and gitlab ci.
My use case is that I need to make sure some files are set before sending it off to codemagic.
I have seen this, but is there some official support around this?
https://stackoverflow.com/questions/58482879/execute-codemagic-build-after-gitlab-pipeline-job-was-executed

Thanks

**Comment by himesh-cm** [comment, marked as answer]
Hi @kyeshmz 

For your particular use case, you can achieve the same via our REST APIs by doing the check at your end for necessary files, and then, if everything is as ok, you can trigger the build via API.

You can find the docs to our APIs here - https://docs.codemagic.io/rest-api/codemagic-rest-api/

Let us know if you need any further assistance.

**Reply by kyeshmz** [reply]
Thanks! I wish there was some kind of sublink inside of the CI document that could link out to that page.
Currently, I don't know which path would actually lead me to that document. If this is something I can PR, will def do so.

**Reply by himesh-cm** [reply]
Hello @kyeshmz 
Here you go, Link for triggering a new build via API - https://docs.codemagic.io/rest-api/builds/#start-a-new-build
